Source/JavaScriptCore/runtime/VMTraps.cpp

```diff
@@ -380,41 +380,19 @@ void VMTraps::willDestroyVM()
 
 CONCURRENT_SAFE void VMTraps::cancelThreadStopIfNeeded()
 {
-    Locker locker { *m_trapSignalingLock };
-
-    // We need to confirm that there are no pending async events before cancelling the
-    // thread stop request. This is because:
-    // 1. The AsyncEvent being cleared may not be the only one needing threads to stop, or
-    // 2. A new AsyncEvent may have been set by another thread before we get here, and we
-    //    we need to keep the thread stop request.
-    if (needHandling(AsyncEvents)) {
-        RELEASE_ASSERT(m_threadStopRequested);
-        return;
-    }
-    if (!m_threadStopRequested)
-        return; // Cancel was already processed due to another thread. Nothing more to do.
+    ASSERT(m_threadStopRequested);
+    ASSERT(!needHandling(AsyncEvents));
```

Comment:
I don't think you can assert this because another thread may have raced you and set the bit before you got into this function.  Remove this ASSERT.